winbuild/build_prepare.py

```diff
@@ -160,18 +159,22 @@ def cmd_msbuild(
         "bins": ["cjpeg.exe", "djpeg.exe"],
     },
     "zlib": {
-        "url": "https://zlib.net/FILENAME",
-        "filename": f"zlib{V['ZLIB_DOTLESS']}.zip",
-        "dir": f"zlib-{V['ZLIB']}",
-        "license": "README",
-        "license_pattern": "Copyright notice:\n\n(.+)$",
+        "url": f"https://github.com/zlib-ng/zlib-ng/archive/refs/tags/{V['ZLIBNG']}.zip",
+        "filename": f"zlib-ng-{V['ZLIBNG']}.zip",
```

Comment:
I've created https://github.com/python-pillow/Pillow/pull/8601 to change this to .tar.gz, so that it matches wheels-dependencies.sh